### PR TITLE
Challenge 1 extension

### LINKED ISSUE #235: Migrate challenge 1 to extension
Use https://github.com/scaffold-eth/se-2-challenges/issues/234 as guide

### PR DESCRIPTION
**Pull request by damianmarti**
Challenge 1 migrated to an extension following #234 

Closes #235

### CODE REVIEW

File: extension/README.md.args.mjs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+> Start your local network (a local instance of a blockchain):
+
+\`\`\`sh
+yarn chain
+\`\`\`
+
+> in a second terminal window, 🛰 deploy your contract (locally):
+
+\`\`\`sh
+cd challenge-1-decentralized-staking
```
Comment (rin-st) [inline extension/README.md.args.mjs:30]: Remove this line

File: extension/README.md.args.mjs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+yarn deploy
+\`\`\`
+
+> in a third terminal window, start your 📱 frontend:
+
+\`\`\`sh
+cd challenge-1-decentralized-staking
```
Comment (rin-st) [inline extension/README.md.args.mjs:37]: Remove this line

File: extension/packages/nextjs/app/page.tsx.args.mjs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+              </a>{" "}
+              !
+            </p>
+          </div>
+        </div>
+      </div>
+    </div>
+`;
+
+export const externalExtensionName = "SpeedRunEthereum Challenge #1";
```
Comment (rin-st) [inline extension/packages/nextjs/app/page.tsx.args.mjs:40]: please add empty line at the end of this file and other files when you see that minus sign. If you use vscode/cursor add this to your settings `"files.insertFinalNewline": true`

**Review by rin-st — COMMENTED** [review]
Lgtm! Added some small changes you need to do

**Comment by damianmarti** [comment]
> Lgtm! Added some small changes you need to do

Thanks!! Changes done!

File: extension/README.md.args.mjs (hunk 2 above)
Comment (damianmarti) [reply extension/README.md.args.mjs]: Removed

File: extension/packages/nextjs/app/page.tsx.args.mjs (hunk 3 above)
Comment (damianmarti) [reply extension/packages/nextjs/app/page.tsx.args.mjs]: Added empty line and added the config to settings. Thanks!

File: extension/packages/nextjs/package.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "humanize-duration": "^3.28.0"
+  }
+}
```
Comment (Pabl0cks) [inline extension/packages/nextjs/package.json:5]: Think we are missing the type:

`"@types/humanize-duration": "^3",`

File: extension/packages/nextjs/app/staker-ui/_components/EthToPrice.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  className?: string;
+};
+
+/**
+ * Display (ETH & USD) value for the input value provided.
+ */
+export const ETHToPrice = ({ value, className = "" }: TBalanceProps) => {
+  const [isEthBalance, setIsEthBalance] = useState(true);
+  const { targetNetwork } = useTargetNetwork();
+  const price = useGlobalState(state => state.nativeCurrencyPrice);
```
[the remark this answers is not in the document]
Comment (damianmarti) [reply extension/packages/nextjs/app/staker-ui/_components/EthToPrice.tsx]: Changed!

File: extension/packages/nextjs/package.json (hunk 4 above)
Comment (damianmarti) [reply extension/packages/nextjs/package.json:8]: Added!

**Review by rin-st — APPROVED** [review]
Thanks! Lgtm!

**Review by Pabl0cks — APPROVED** [review]
LGTM! Merging 🙏

**Pull request merged**